src/Features/Core/Portable/ChangeSignature/DelegateInvokeMethodReferenceFinder.cs

```diff
@@ -72,58 +72,49 @@ protected override bool CanFind(IMethodSymbol symbol)
 
         protected override async ValueTask<ImmutableArray<FinderLocation>> FindReferencesInDocumentAsync(
             IMethodSymbol methodSymbol,
-            HashSet<string>? globalAliases,
-            Document document,
-            SemanticModel semanticModel,
-            FindReferenceCache cache,
+            FindReferencesDocumentState state,
             FindReferencesSearchOptions options,
             CancellationToken cancellationToken)
         {
             // FAR on the Delegate type and use those results to find Invoke calls
 
-            var syntaxFactsService = document.GetRequiredLanguageService<ISyntaxFactsService>();
-            var semanticFactsService = document.GetRequiredLanguageService<ISemanticFactsService>();
+            var syntaxFacts = state.SyntaxFacts;
+            var semanticFacts = state.SemanticFacts;
 
-            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var root = state.Root;
```

Comment:
state object also contains the most common doc-oriented state needed by practically all finding methods.